sdk/storage/Azure.Storage.Blobs/src/Sas/KeySasQueryParametersExtensions.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace Azure.Storage.Sas
+{
+    internal static class KeySasQueryParametersExtensions
+    {
+        /// <summary>
+        /// Parses the key properties into the QueryParameters instance.
+        /// </summary>
+        /// <param name="parameters">
+        /// The BlobSasQueryParameters or DataLakeSasQueryParameters instance.
+        /// </param>
+        /// <param name="values">
+        /// Dictionary of keys and values.
+        /// </param>
+        internal static void ParseKeyProperties(
+            this
+#if BlobSDK
+            BlobSasQueryParameters
+#elif DataLakeSDK
+            DataLakeSasQueryParameters
+#endif
+            parameters,
+            Dictionary<string, string> values)
+        {
+            // make copy, otherwise we'll get an exception when we remove
+            IEnumerable<KeyValuePair<string, string>> kvps = values.ToArray();
+            var isSasKey = true;
+            parameters._keyProperties = new UserDelegationKeyProperties();
```

Comment:
Nit - make it a `KeyProperties` property instead of an internal field.

Reply:
Fixed